ci/docker/Dockerfile.build.ubuntu_build_cuda110

```diff
@@ -0,0 +1,46 @@
+# -*- mode: dockerfile -*-
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# Dockerfile to build MXNet on Ubuntu 16.04 for GPU but on
+# a CPU-only instance. This restriction is caused by the CPP-
+# package generation, requiring the actual CUDA library to be
+# present
+
+FROM nvidia/cuda:11.0-cudnn8-devel-ubuntu16.04
+
+WORKDIR /work/deps
+
+COPY install/ubuntu_core.sh /work/
+RUN /work/ubuntu_core.sh
+COPY install/deb_ubuntu_ccache.sh /work/
+RUN /work/deb_ubuntu_ccache.sh
+COPY install/ubuntu_python.sh /work/
+COPY install/requirements /work/
+RUN /work/ubuntu_python.sh
+
+# Keep this at the end since this command is not cachable
+ARG USER_ID=0
+ARG GROUP_ID=0
+COPY install/ubuntu_adduser.sh /work/
+RUN /work/ubuntu_adduser.sh
+
+COPY runtime_functions.sh /work/
+
+WORKDIR /work/mxnet
+ENV LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/local/cuda/compat
```

Comment:
I remember facing issues with cuda/compat directory while migrating from G3 to G4. Do we need this? @ptrendx @leezu confirm.
It was only TVM OP which needed that right? And since we are disabling it we shouldn't need compat dir.

Correct me if wrong.

Reply:
Pretty sure we need this:
```
/usr/local/cuda-10.1/compat
[root@514f7b276090 compat]# ls
libcuda.so    libcuda.so.418.152.00                    libnvidia-ptxjitcompiler.so.1
libcuda.so.1 
```

Reply:
Without this, the cpp-package fails to build (due to unable to find libcuda.so.1, which libmxnet.so is linked against.)

I could also disable the cpp-package portion of the build, since it's actually not being used in the test pipeline steps.